examples/search_examples/public/components/app.tsx

```diff
@@ -0,0 +1,173 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage, I18nProvider } from '@kbn/i18n/react';
+import { BrowserRouter as Router } from 'react-router-dom';
+
+import {
+  EuiButton,
+  EuiPage,
+  EuiPageBody,
+  EuiPageContent,
+  EuiPageContentBody,
+  EuiPageHeader,
+  EuiTitle,
+  EuiText,
+  EuiFlexGrid,
+  EuiFlexItem,
+} from '@elastic/eui';
+
+import { CoreStart } from '../../../../src/core/public';
+import { NavigationPublicPluginStart } from '../../../../src/plugins/navigation/public';
+
+import { PLUGIN_ID, PLUGIN_NAME, IMyStrategyResponse } from '../../common';
+
+import {
+  DataPublicPluginStart,
+  IndexPatternSelect,
+  IEsSearchResponse,
+  IIndexPattern,
+} from '../../../../src/plugins/data/public';
+
+interface SearchExamplesAppDeps {
+  basename: string;
+  notifications: CoreStart['notifications'];
+  savedObjectsClient: CoreStart['savedObjects']['client'];
+  navigation: NavigationPublicPluginStart;
+  data: DataPublicPluginStart;
+}
+
+export const SearchExamplesApp = ({
+  basename,
+  notifications,
+  savedObjectsClient,
+  navigation,
+  data,
+}: SearchExamplesAppDeps) => {
+  const [result, setResult] = useState<IEsSearchResponse | undefined>();
+  const [indexPatternId, setIndexPatternId] = useState<string>();
+
+  const doAsyncSearch = async (strategy?: string) => {
+    let indexPattern: IIndexPattern | undefined;
+    if (indexPatternId) {
+      indexPattern = await data.indexPatterns.get(indexPatternId);
+    }
+    const request = {
+      params: {
+        index: indexPattern ? indexPattern.title : '',
+        body: {
+          aggs: {
+            avg_bytes: { avg: { field: 'bytes' } },
+          },
+        },
+      },
+    };
+    const search$ = data.search.search(request, {
+      strategy,
+    });
+    search$.subscribe((response) => {
+      // Need to add is running and is partial to oss
+      if (!response.isPartial && !response.isRunning) {
+        setResult(response);
+        notifications.toasts.addSuccess(
+          `Searched ${response.rawResponse.hits.total} documents. Result is ${
+            response.rawResponse.aggregations.avg_bytes.value
+          }. Is this Cool? ${(response as IMyStrategyResponse).cool}`
+        );
+      }
+    });
+  };
+
+  const onClickHandler = () => {
+    doAsyncSearch();
+  };
+
+  const onMyStrategyClickHandler = () => {
+    doAsyncSearch('myStrategy');
+  };
+
+  return (
+    <Router basename={basename}>
+      <I18nProvider>
+        <>
+          <navigation.ui.TopNavMenu appName={PLUGIN_ID} showSearchBar={true} />
```

Comment:
Right now it shows the date picker but it doesn't apply the filter to the query... Should we remove that?

Reply:
I plan to integrate it!

Reply:
Did it :)